dtfs-central-api/api-tests/v1/tfm/deal/tfm-bss-deals-put-submit.api-test.js

```diff
@@ -29,6 +30,30 @@ describe('/v1/tfm/deals/submit - BSS/EWCS deal', () => {
     ]);
   });
 
+  describe('auditting', () => {
```

Comment:
I usually like not to have any describe block here -- see `post-create-user.api-tests.js` -- the reason why is I like everything to be contained inside the test helper itself (with it's own describe block)

We could avoid this here by extracting this create a deal to the main before each (as it is needed in all tests it seems).

If, for another set of tests, this isnt possible, this probably implies our test formatting is wrong.... So say you have 

```
describe('/v1/tfm/deals/submit - BSS/EWCS deal', () => {
  beforeEach(async () => {
    await wipeDB.wipe([
      CONSTANTS.DB_COLLECTIONS.DEALS,
      CONSTANTS.DB_COLLECTIONS.FACILITIES,
      CONSTANTS.DB_COLLECTIONS.TFM_DEALS,
      CONSTANTS.DB_COLLECTIONS.TFM_FACILITIES,
    ]);
  });

  describe('auditting', () => {
    let dealId;

    beforeEach(async () => {
      const { body: createDealBody } = await api
        .post({
          deal: newDeal,
          user: MOCK_PORTAL_USER,
        })
        .to('/v1/portal/deals');

      dealId = createDealBody._id;
    });

    itValidatesAuditDetails({...});
  });

  it('returns dealSnapshot with tfm object', async () => {
    ...
  });

  it('does something that doesn't require dealID')

```

we could do:
```
describe('/v1/tfm/deals/submit - BSS/EWCS deal', () => {
  beforeEach(async () => {
    await wipeDB.wipe([
      CONSTANTS.DB_COLLECTIONS.DEALS,
      CONSTANTS.DB_COLLECTIONS.FACILITIES,
      CONSTANTS.DB_COLLECTIONS.TFM_DEALS,
      CONSTANTS.DB_COLLECTIONS.TFM_FACILITIES,
    ]);
  });

    describe('when a valid deal id is provided', () => {
        let dealId;

        beforeEach(async () => {
        const { body: createDealBody } = await api
            .post({
            deal: newDeal,
            user: MOCK_PORTAL_USER,
            })
            .to('/v1/portal/deals');

        dealId = createDealBody._id;
        });

        itValidatesAuditDetails({...});

        it('returns dealSnapshot with tfm object', async () => {
            ...
        });
    });

    describe('when a valid deal id is provided', () => {
        it('does something that doesn't require dealID')
    })
```